datetime/datetime_test.go

```diff
@@ -0,0 +1,262 @@
+package datetime_test
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	. "github.com/tarantool/go-tarantool"
+	"github.com/tarantool/go-tarantool/test_helpers"
+	"gopkg.in/vmihailenco/msgpack.v2"
+)
+
+// There is no way to skip tests in testing.M,
+// so we use this variable to pass info
+// to each testing.T that it should skip.
+var isDatetimeSupported = false
+
+var server = "127.0.0.1:3013"
+var opts = Opts{
+	Timeout: 500 * time.Millisecond,
+	User:    "test",
+	Pass:    "test",
+}
+
+var space = "testDatetime"
+var index = "primary"
+
+type TupleDatetime struct {
+	tm time.Time
+}
+
+func (t *TupleDatetime) DecodeMsgpack(d *msgpack.Decoder) error {
+	var err error
+	var l int
+	if l, err = d.DecodeSliceLen(); err != nil {
+		return err
+	}
+	if l != 1 {
+		return fmt.Errorf("array len doesn't match: %d", l)
+	}
+
+	res, err := d.DecodeInterface()
+	if err != nil {
+		return err
+	}
+	t.tm = res.(time.Time)
+
+	return nil
+}
+
+func connectWithValidation(t *testing.T) *Connection {
+	conn, err := Connect(server, opts)
+	if err != nil {
+		t.Errorf("Failed to connect: %s", err.Error())
+	}
+	if conn == nil {
+		t.Errorf("conn is nil after Connect")
+	}
+	return conn
+}
+
+func tupleValueIsDatetime(t *testing.T, tuples []interface{}, tm time.Time) {
+	if tpl, ok := tuples[0].([]interface{}); !ok {
+		t.Errorf("Unexpected return value body")
+	} else {
+		if len(tpl) != 1 {
+			t.Errorf("Unexpected return value body (tuple len = %d)", len(tpl))
+		}
+		if val, ok := tpl[0].(time.Time); !ok || val != tm {
+			fmt.Println("Tuple:   ", tpl[0])
+			fmt.Println("Expected:", val)
+			t.Errorf("Unexpected return value body (tuple 0 field)")
+		}
+	}
+}
+
+func tupleInsertSelectDelete(t *testing.T, conn *Connection, datetime string) {
+	tm, err := time.Parse(time.RFC3339, datetime)
+	if err != nil {
+		t.Errorf("Time (%s) parse failed: %s", datetime, err)
```

Comment:
It seems that Errorf not finished the execution, see https://github.com/tarantool/go-tarantool/issues/150

Reply:
Updated.

```diff
diff --git a/datetime/datetime_test.go b/datetime/datetime_test.go
index 9be9b40..d7c202d 100644
--- a/datetime/datetime_test.go
+++ b/datetime/datetime_test.go
@@ -40,7 +40,7 @@ func (t *TupleDatetime) DecodeMsgpack(d *msgpack.Decoder) error {
 		return err
 	}
 	if l != 1 {
-		return fmt.Errorf("array len doesn't match: %d", l)
+		return fmt.Fatalf("array len doesn't match: %d", l)
 	}
 
 	res, err := d.DecodeInterface()
@@ -55,25 +55,25 @@ func (t *TupleDatetime) DecodeMsgpack(d *msgpack.Decoder) error {
 func connectWithValidation(t *testing.T) *Connection {
 	conn, err := Connect(server, opts)
 	if err != nil {
-		t.Errorf("Failed to connect: %s", err.Error())
+		t.Fatalf("Failed to connect: %s", err.Error())
 	}
 	if conn == nil {
-		t.Errorf("conn is nil after Connect")
+		t.Fatalf("conn is nil after Connect")
 	}
 	return conn
 }
 
 func tupleValueIsDatetime(t *testing.T, tuples []interface{}, tm time.Time) {
 	if tpl, ok := tuples[0].([]interface{}); !ok {
-		t.Errorf("Unexpected return value body")
+		t.Fatalf("Unexpected return value body")
 	} else {
 		if len(tpl) != 1 {
-			t.Errorf("Unexpected return value body (tuple len = %d)", len(tpl))
+			t.Fatalf("Unexpected return value body (tuple len = %d)", len(tpl))
 		}
 		if val, ok := tpl[0].(time.Time); !ok || val != tm {
 			fmt.Println("Actual tuple:  ", val)
 			fmt.Println("Expected tuple:", tm)
-			t.Errorf("Unexpected return value body (tuple 0 field)")
+			t.Fatalf("Unexpected return value body (tuple 0 field)")
 		}
 	}
 }
@@ -82,7 +82,7 @@ func TestDatetimeEncodeDecode(t *testing.T) {
 	t.Skip("not implemented")
 	tm1, err := time.Parse(time.RFC3339, "2006-01-02T15:04:05Z")
 	if err != nil {
-		t.Errorf("Parse is failed %s", err)
+		t.Fatalf("Parse is failed %s", err)
 	}
 
 	// Encode
@@ -91,31 +91,31 @@ func TestDatetimeEncodeDecode(t *testing.T) {
 	e := msgpack.NewEncoder(bufferWriter)
 	err = e.Encode(tm1)
 	if err != nil {
-		t.Errorf("Datetime encode failed %s", err)
+		t.Fatalf("Datetime encode failed %s", err)
 	}
 
 	// Decode
 	dec := msgpack.NewDecoder(&buf)
 	tm2, err := dec.DecodeInterface()
 	if err != nil {
-		t.Errorf("Datetime decode failed %s", err)
+		t.Fatalf("Datetime decode failed %s", err)
 	}
 
 	if tm1 != tm2 {
-		t.Errorf("Datetime is not the same as encoded ('%s' and '%s')", tm1, tm2)
+		t.Fatalf("Datetime is not the same as encoded ('%s' and '%s')", tm1, tm2)
 	}
 }
 
 func tupleInsertSelectDelete(t *testing.T, conn *Connection, datetime string) {
 	tm, err := time.Parse(time.RFC3339, datetime)
 	if err != nil {
-		t.Errorf("Time (%s) parse failed: %s", datetime, err)
+		t.Fatalf("Time (%s) parse failed: %s", datetime, err)
 	}
 
 	// Insert tuple with datetime.
 	resp, err := conn.Insert(space, []interface{}{tm})
 	if err != nil {
-		t.Errorf("Datetime insert failed: %s", err.Error())
+		t.Fatalf("Datetime insert failed: %s", err.Error())
 	}
 	fmt.Println(resp)
 
@@ -124,20 +124,20 @@ func tupleInsertSelectDelete(t *testing.T, conn *Connection, datetime string) {
 	var limit uint32 = 1
 	resp, err = conn.Select(space, index, offset, limit, IterEq, []interface{}{tm})
 	if err != nil {
-		t.Errorf("Datetime select failed: %s", err.Error())
+		t.Fatalf("Datetime select failed: %s", err.Error())
 	}
 	if resp == nil {
-		t.Errorf("Response is nil after Select")
+		t.Fatalf("Response is nil after Select")
 	}
 	tupleValueIsDatetime(t, resp.Data, tm)
 
 	// Delete tuple with datetime.
 	resp, err = conn.Delete(space, index, []interface{}{tm})
 	if err != nil {
-		t.Errorf("Datetime delete failed: %s", err.Error())
+		t.Fatalf("Datetime delete failed: %s", err.Error())
 	}
 	if resp == nil {
-		t.Errorf("Response is nil after Delete")
+		t.Fatalf("Response is nil after Delete")
 	}
 	tupleValueIsDatetime(t, resp.Data, tm)
 }
@@ -221,31 +221,31 @@ func TestDatetimeReplace(t *testing.T) {
 
 	tm, err := time.Parse(time.RFC3339, "2007-01-02T15:04:05Z")
 	if err != nil {
-		t.Errorf("Time parse failed: %s", err)
+		t.Fatalf("Time parse failed: %s", err)
 	}
 
 	resp, err := conn.Replace(space, []interface{}{tm})
 	if err != nil {
-		t.Errorf("Datetime replace failed: %s", err)
+		t.Fatalf("Datetime replace failed: %s", err)
 	}
 	if resp == nil {
-		t.Errorf("Response is nil after Replace")
+		t.Fatalf("Response is nil after Replace")
 	}
 	tupleValueIsDatetime(t, resp.Data, tm)
 
 	resp, err = conn.Select(space, index, 0, 1, IterEq, []interface{}{tm})
 	if err != nil {
-		t.Errorf("Datetime select failed: %s", err)
+		t.Fatalf("Datetime select failed: %s", err)
 	}
 	if resp == nil {
-		t.Errorf("Response is nil after Select")
+		t.Fatalf("Response is nil after Select")
 	}
 	tupleValueIsDatetime(t, resp.Data, tm)
 
 	// Delete tuple with datetime.
 	resp, err = conn.Delete(space, index, []interface{}{tm})
 	if err != nil {
-		t.Errorf("Datetime delete failed: %s", err.Error())
+		t.Fatalf("Datetime delete failed: %s", err.Error())
 	}
 }
 ```